lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -341,6 +345,18 @@ getInitial = label "getInitial" $ do
                                 Transactions
 -------------------------------------------------------------------------------}
 
+-- | Glues together a tx, witnesses into a full @SignedTxBinary@.
+--
+-- FIXME: The txId in the Tx relies on the full fragment, and to construct the
+-- witnesses the inputs and outputs need to be serialised to binary. It might be
```

Comment:
I don't understand this comment? May you clarify what problems is it trying to highlight?

Reply:
This function gives us the power to construct a tx with an arbitrary txId and the wrong number of witnesses (which themselves also may be wrong).

I think we very rarely need that power (maybe external tx tests).

I think a `ins -> outs -> keyLookup -> SignedTxBinary` constructor would be better (as in #1021 ).

Reply:
Updated